### PR TITLE
[editor, editor-common] fix: add types annotations

### PR DESCRIPTION
**Pull request by maimonav**
Add missing types in `createPlugins` and `getModalStyles`.

### CODE REVIEW

**Comment by shaulgo** [comment]
@mharsat please take a look

File: packages/editor/web/src/RichContentEditor/createPlugins.ts [hunk 1]
```diff
@@ -42,7 +42,13 @@ const createPlugins = ({
   plugins?: CreatePluginFunction[];
   context: EditorContextType;
   commonPubsub: Pubsub;
-}) => {
+}): {
+  pluginInstances: unknown[];
```
Comment (mharsat) [inline packages/editor/web/src/RichContentEditor/createPlugins.ts:46]: I suggest
```tsx
pluginInstances: (EditorPlugin | ReturnType<CreatePluginFunction>)[]
```

Or more specifically `EditorPlugin & { decorator }` for the first part

Comment (maimonav) [reply packages/editor/web/src/RichContentEditor/createPlugins.ts:46]: Not all plugins has & { decorator },
I can go with
```suggestion
  pluginInstances: (
    | EditorPlugin
    | (EditorPlugin & { decorator })
    | ReturnType<CreatePluginFunction>
  )[];
```
Or with your first suggestion.

Comment (mharsat) [reply packages/editor/web/src/RichContentEditor/createPlugins.ts:46]: Looks good, I'd make it optional like this

```tsx
pluginInstances: (
    | (EditorPlugin & { decorator?: DraftDecorator })
    | ReturnType<CreatePluginFunction>
  )[]
```

**Review by mharsat — APPROVED** [review]

File: packages/editor/web/src/RichContentEditor/createPlugins.ts (hunk 1 above)
Comment (maimonav) [reply packages/editor/web/src/RichContentEditor/createPlugins.ts:46]: Great! thanks.

**Pull request merged**